Varying weirs patch

krober10nd opened the pull request:
This adds support for specifying the resolution along the face of the weir in meters. The spacing along the weirs is the fourth column of each entry in the cell-array. 

This patch also will ensure that edge constraints are preserved when using the plus operator so that the structure of the weirs is not destroyed.

Commit:
Fourth column of cell-array describing weir denotes the spacing
-Spacing on face of weir can be variable specified by user

Commit:
Support for the perservation of edge constraints when plusing

Commit:
Changes to support specification of spacing along weir face.

CHLNDDEV commented:
I still need to carry out the fixed points to the new merged mesh class. I should be able to finish that today.

WPringle reviewed (approved):
I'm happy with this.. Only major changes to weirs, and using the egfix in the delaunay triangulation when merging meshes.

Commit:
Ensuring point and edge constraints are perserved in plus operation
-Plus operator will perserve point and edge constraints in either
first (and) second meshes being merged and then return these combined points
in the msh.pfix and msh.egfix fields.
